proxy.go

```diff
@@ -321,48 +321,59 @@ func (rp *reverseProxy) serveFromCache(s *scope, srw *statResponseWriter, req *h
 	reader := bufferedRespWriter.Reader()
 	contentMetadata := cache.ContentMetadata{Length: contentLength, Encoding: contentEncoding, Type: contentType}
 
-	if bufferedRespWriter.StatusCode() != http.StatusOK || s.canceled {
-		// Do not cache non-200 or cancelled responses.
-		// Restore the original status code by proxyRequest if it was set.
-		if srw.statusCode != 0 {
-			bufferedRespWriter.WriteHeader(srw.statusCode)
-		}
-
-		// mark transaction as failed
-		// todo: discuss if we should mark it as failed upon timeout. The rational against it would be to hope that
-		// 		 partial results of the query are cached and therefore subsequent execution can succeed
-		if err = userCache.Fail(key); err != nil {
-			log.Errorf("%s: %s; query: %q", s, err, q)
-		}
-
+	toCache := isToCache(contentLength, s)
+	if !toCache {
```

Comment:
Looking at the content of the method `isToCache`, I think we can inline it
```suggestion
	if length > s.user.cache.MaxPayloadSize {
```